### PR TITLE
fix httpclient

### PR DESCRIPTION
**Pull request by gev**
Fix http client. Truncate and close stream

### CODE REVIEW

File: extensions/modules/src/org/exist/xquery/modules/httpclient/BaseHTTPClientFunction.java [hunk 1]
```diff
@@ -380,11 +380,10 @@ public String getCacheClass() {
                         final byte buf[] = new byte[4096];
                         int read = -1;
                         while((read = cfis.read(buf)) > -1) {
-                            baos.write(buf);
+                            baos.write(buf, 0, read);
                         }
-                        
-                        builder.characters(URLEncoder.encode(EncodingUtil.getString(baos.toByteArray(), ((HttpMethodBase)method).getResponseCharSet()), "UTF-8"));
                         baos.close();
+                        builder.characters(URLEncoder.encode(EncodingUtil.getString(baos.toByteArray(), ((HttpMethodBase)method).getResponseCharSet()), "UTF-8"));
```
Comment (dizzzz) [inline extensions/modules/src/org/exist/xquery/modules/httpclient/BaseHTTPClientFunction.java]: is this really required?

**Comment by dizzzz** [comment]
Please could you describe what the fix is about?

File: extensions/modules/src/org/exist/xquery/modules/httpclient/BaseHTTPClientFunction.java (hunk 1 above)
Comment (shabanovd) [inline extensions/modules/src/org/exist/xquery/modules/httpclient/BaseHTTPClientFunction.java]: it was there before

**Comment by shabanovd** [comment]
API misuse here, `baos.write(buf);` writing all bytes from array, but it may read just some. That's ehy  `baos.write(buf, 0, read);` must be used.

**Pull request merged**